test/integration/devise_confirmation_test.rb

```diff
@@ -0,0 +1,63 @@
+require 'test_helper'
+
+class DeviseConformatinTest < ActionDispatch::IntegrationTest
+
+  test "Confirmation page requires password" do
+    email = 'newuser-fc591ba5dac844d0@example.com'
+    name = 'newuser-fc591ba5dac844d0'
+  
+    post user_registration_path,
+      user: { name: name, email: email, password: 'Test1234', password_confirmation: 'Test1234' }
+    assert_response :redirect
+    
+    u = ::User.where(email: email).first()
+    token = u.confirmation_token
+
+    get '/users/confirmation', {confirmation_token: token}
+    assert_response :success
+
+    u = ::User.where(email: email).first()
+    assert_not u.confirmed?
+  end
+
+  test "Confirmation completes after password verification" do
+    email = 'newuser-fc591ba5dac844d0@example.com'
+    name = 'newuser-fc591ba5dac844d0'
+    password = 'Test1234'
```

Comment:
I think we should use a randomly generated password instead of hardcoding else secrets scanning tools will start complaining.

Reply:
done